experimental/InjectiveResolutions/src/LocalCohomology.jl

```diff
@@ -0,0 +1,642 @@
+    valid = true # is one strip empty? => the intersection is empty
+    A_tuple = []
+    for i in eachindex(tau)
+      if all(x -> x == -Inf*one(field), tau[i])
+        continue
+      end
+      if stripes[i][tuple[i]] isa Polyhedron
+        push!(_delta, stripes[i][tuple[i]])
+      else #stripe is empty
+        valid = false
+        break
+      end
+
+      # compute the set of all indices j such that the current stripe lies in J^j 
+      A_i = Vector{Int}()
+      if sorted_taus[i][tuple[i]] == -Inf*one(field)
+        A_i = findall(x -> x == -Inf*one(field), tau[i])
+      else
+        A_i = findall(x -> x == -Inf*one(field) || (x <= sorted_taus[i][tuple[i]]), tau[i])
+      end
+      push!(A_tuple, A_i)
+    end
+    if length(_delta) > 0 && valid
+      # compute the sector \Delta(l1,...,ln)
+      poly_tuple = intersect(_delta...)
+      if dim(poly_tuple) < 0
+        continue
+      elseif dim(poly_tuple) == 0 && length(lattice_points(poly_tuple)) == 0
+        continue
+      end
+
+      # get A \subseteeq {1,...,r} such that |Delta(l1,...,ln) \in J^j if and only if j \in A
+      A = intersect(A_tuple...)
+
+      # compute local cohomology of sector
+      H_A, A_ = _local_cohomology_sector(field, A, j, k, phi, psi)
+
+      #avoid duplicates 
+      s_A = [s for s in S_A if s.A == A_ && s.sector == poly_tuple]
+      if is_empty(s_A)
+        push!(S_A, SectorLC(A_, poly_tuple, collect(tuple), H_A)) # changed
+      end
+    end
+  end
+  return S_A
+end
+
+@doc raw"""
+  _local_cohomology_sector(field::Field,A::Vector{Int},j::Integer,k::Integer,phi::Union{Vector{Any},MatElem{T}},psi::Union{Vector{Any},MatElem{T}}) where {T<:FieldElem}
+
+This function follows the construction in the proof of [HM05](@cite) (Theorem 5.2).
+"""
+function _local_cohomology_sector(
+  field::Field,
+  A::Vector{Int},
+  j::Integer,
+  k::Integer,
+  phi::MatElem{T},
+  psi::MatElem{T}
+) where {T<:FieldElem}
+  @assert elem_type(field) == T
+  # divide A into triple
+  A_0 = [a for a in A if a <= j]
+  A_1 = [a for a in A if j < a <= k]
+  A_2 = [a for a in A if k < a]
+  _A = [A_0, A_1, A_2]
+
+  # define vector spaces J_{S_A0}, J_{S_A1} and J_{S_A2}
+  F_0 = free_module(field, length(A_0))
+  F_1 = free_module(field, length(A_1))
+  F_2 = free_module(field, length(A_2))
+
+  #compute the maps by deleting rows and columns in phi and psi
+  #phi
+  phi_del = matrix(field, zeros(field, length(A_0), length(A_1)))
+
+  if !is_empty(phi) # `is_empty` is true for the the m x 0-matrix
+    rows = []
+    for i in A_0 #delete i-th row of phi for i \notin A_0
+      push!(rows, phi[i, :])
+    end
+    _phi_del = transpose(hcat(rows...))
+
+    if all(>(0), size(_phi_del)) #not of size 0xn or nx0
+      columns = []
+      for i in A_1 #delete i-th column of phi for i \notin A_1
+        push!(columns, _phi_del[:, i - j])
+      end
+      if length(columns) > 0
+        phi_del = matrix(field, hcat(columns...))
+      end
```

Comment:
Again, I may have missed something, but perhaps this also works?
```suggestion
      if length(A_1) > 0
        phi_del = matrix(field, _phi_del[:, A_1 - j])
      end
```